pkg/storage/snapshot/restore.go

```diff
@@ -128,20 +123,35 @@ func (ctrl *VMRestoreController) updateVMRestore(vmRestoreIn *snapshotv1.Virtual
 	if vmRestoreDeleting(vmRestoreOut) {
 		return 0, ctrl.handleVMRestoreDeletion(vmRestoreOut, target)
```

Comment:
Calling handleVMRestoreDeletion without an initialized Status is problematic since we are adding conditions in that function and the status can be nil